pkg/services/ngalert/remote/remote_primary_forked_alertmanager.go

```diff
@@ -2,32 +2,70 @@ package remote
 
 import (
 	"context"
+	"fmt"
+	"sync"
 
+	"github.com/grafana/grafana/pkg/infra/log"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	"github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
 )
 
 type RemotePrimaryForkedAlertmanager struct {
+	log log.Logger
+	mtx sync.Mutex
+
 	internal notifier.Alertmanager
-	remote   notifier.Alertmanager
+	remote   remoteAlertmanager
+
+	currentConfigHash string
 }
 
-func NewRemotePrimaryForkedAlertmanager(internal, remote notifier.Alertmanager) *RemotePrimaryForkedAlertmanager {
+func NewRemotePrimaryForkedAlertmanager(log log.Logger, internal notifier.Alertmanager, remote remoteAlertmanager) *RemotePrimaryForkedAlertmanager {
 	return &RemotePrimaryForkedAlertmanager{
+		log:      log,
 		internal: internal,
 		remote:   remote,
 	}
 }
 
+// ApplyConfig will send the configuration to the remote Alertmanager on startup and on change.
+// The call is always first delegated to the internal Alertmanager.
 func (fam *RemotePrimaryForkedAlertmanager) ApplyConfig(ctx context.Context, config *models.AlertConfiguration) error {
+	if err := fam.internal.ApplyConfig(ctx, config); err != nil {
+		return fmt.Errorf("failed to call ApplyConfig on the internal Alertmanager: %w", err)
+	}
+
+	fam.mtx.Lock()
+	defer fam.mtx.Unlock()
```

Comment:
I might be acquiring a lock for a long time, but wasn't sure whether to take this simpler approach or to use a `RWMutex` to lock/unlock on every potential read/write to `currentConfigHash`